### PR TITLE
Cpupower monitor

### PR DESCRIPTION
**Pull request by kalshett**

### CODE REVIEW

**Comment by kalshett** [comment]
[root@ltcden14-lp10 cpu]# cat x

avocado run cpupower_monitor.py


[root@ltcden14-lp10 cpu]# ./x
Fetching asset from cpupower_monitor.py:CpupowerMonitor.test_workload
JOB ID     : df7b354c679b6c11501ade40dfb232c5f8772739
JOB LOG    : /home/kal/avocado-fvt-wrapper/results/job-2021-06-23T02.54-df7b354/job.log
 (1/4) cpupower_monitor.py:CpupowerMonitor.test_workload: PASS (108.24 s)
 (2/4) cpupower_monitor.py:CpupowerMonitor.test_disable_idlestate: PASS (8.35 s)
 (3/4) cpupower_monitor.py:CpupowerMonitor.test_idlestate_smt: FAIL: cpus have not entered idle states after killing ebizzy workload (121.88 s)
 (4/4) cpupower_monitor.py:CpupowerMonitor.test_idlestate_single_core: PASS (150.21 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/kal/avocado-fvt-wrapper/results/job-2021-06-23T02.54-df7b354/results.html
JOB TIME   : 391.07 s

One test case failure is expected as even if we stop workload some CPUs may have stil show values as zero, hence test case failed.

**Review by nasastry — APPROVED** [review]
LGTM

File: cpu/cpupower_monitor.py [hunk 1]
```diff
@@ -41,7 +41,14 @@ def setUp(self):
             if 'Available idle states: ' in line:
                 self.states_list = (line.split('Available idle states: ')[-1]).split()
                 break
-        self.log.info("Idle states on the system are: ", self.states_list)
+        self.log.info("Idle states on the system are: %s", self.states_list)
```
Comment (PraveenPenguin) [inline cpu/cpupower_monitor.py:44]: as using `%s` and `,` will not make substitution , either go with older way if still want to change please use this way 

`self.log.info("Idle states on the system are: %s" %self.states_list)`

as looking this opportunity i recommend to use string.format() using that we can get rid of all these issue

File: cpu/cpupower_monitor.py [hunk 2]
```diff
@@ -41,7 +41,14 @@ def setUp(self):
...
+
+        for line in output.splitlines():
+            if 'Number of idle states: ' in line:
+                self.states_tot = int(line.split('Number of idle states: ')[1])
+                break
+        self.log.info("Total Idle states: %d", self.states_tot)
```
Comment (PraveenPenguin) [inline cpu/cpupower_monitor.py:50]: same here

File: cpu/cpupower_monitor.py [hunk 3]
```diff
@@ -41,7 +41,14 @@ def setUp(self):
...
+
```
Comment (PraveenPenguin) [inline cpu/cpupower_monitor.py:51]: do not use white space in same block of code

File: cpu/cpupower_monitor.py [hunk 4]
```diff
@@ -88,14 +95,14 @@ def test_workload(self):
                                  shell=True)
         obj.start()
         time.sleep(2)
-        for i in range(len(self.states_list)):
+        for i in range(self.states_tot - 1):
```
Comment (PraveenPenguin) [inline cpu/cpupower_monitor.py:98]: ok

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

File: cpu/cpupower_monitor.py (hunk 1 above)
Comment (kalshett) [reply cpu/cpupower_monitor.py]: Done.

File: cpu/cpupower_monitor.py (hunk 2 above)
Comment (kalshett) [reply cpu/cpupower_monitor.py]: Done

File: cpu/cpupower_monitor.py (hunk 3 above)
Comment (kalshett) [reply cpu/cpupower_monitor.py]: Done

**Comment by kalshett** [comment]
@PraveenPenguin I have resent the PR, please review and merge.

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Pull request merged**